File: app/src/main/java/org/deiverbum/app/data/entity/BiblieBookEntity.java
package org.deiverbum.app.data.entity;

import static org.deiverbum.app.utils.Constants.BIBLE_BOOK;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import org.deiverbum.app.model.BibleBook;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2022.2
 */

@Entity(tableName = BIBLE_BOOK)

public class BiblieBookEntity {
    @NonNull
    @PrimaryKey
    @ColumnInfo(name = "bookID")
    public Integer libroId = 0;

    @NonNull
    @ColumnInfo(name = "bookType")
    public Integer bookType = 0;

    @NonNull
    @ColumnInfo(name = "shortName")
    public String shortName = "";

    @NonNull
    @ColumnInfo(name = "longName")
    public String longName = "";

    @NonNull
    @ColumnInfo(name = "liturgyName")
    public String liturgyName = "";

    @NonNull
    @ColumnInfo(name = "orderName")
    public String orderName = "";

    @NonNull
    public String getShortName() {
        return shortName;
    }

    @NonNull
    public String getLongName() {
        return longName;
    }

    @NonNull
    public String getLiturgyName() {
        return liturgyName;
    }

    public BibleBook getDomainModel() {
        BibleBook theModel = new BibleBook();
        theModel.setName(getLongName());
        theModel.setLiturgyName(getLiturgyName());
        theModel.setShortName(getShortName());
        return theModel;
    }

}

File: app/src/main/java/org/deiverbum/app/data/entity/BiblieBookEntity.kt
package org.deiverbum.app.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import org.deiverbum.app.model.BibleBook
import org.deiverbum.app.util.Constants

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Entity(tableName = Constants.BIBLE_BOOK)
class BiblieBookEntity {
    @JvmField
    @PrimaryKey
    @ColumnInfo(name = "bookID")
    var libroId = 0

    @JvmField
    @ColumnInfo(name = "bookType")
    var bookType = 0

    @JvmField
    @ColumnInfo(name = "shortName")
    var shortName = ""

    @JvmField
    @ColumnInfo(name = "longName")
    var longName = ""

    @JvmField
    @ColumnInfo(name = "liturgyName")
    var liturgyName = ""

    @JvmField
    @ColumnInfo(name = "orderName")
    var orderName = ""
    val domainModel: BibleBook
        get() {
            val theModel = BibleBook()
            theModel.name = longName
            theModel.liturgyName = liturgyName
            theModel.shortName = shortName
            return theModel
        }
}